lib/src/providers/TranslationProvider/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { PropsWithChildren, useMemo } from 'react';
+import { createIntl, createIntlCache, RawIntlProvider, type ResolvedIntlConfig } from 'react-intl';
+import resourceLoader, { type ResourceLoader } from '../../utils/suspense/resourceLoader';
+import { VERNA_SUPPORTED_LOCALES } from '../../settings';
+import { TranslationType } from '../../types/translations';
+
+interface TranslationProviderProps {
+  defaultLocale?: string;
+  intl?: ResolvedIntlConfig;
+  locale?: string;
+  schemaTranslations?: TranslationType;
+}
+
+type MessageLoader = ResourceLoader<ResolvedIntlConfig['messages']>;
+type MessageLoaders = Record<string, MessageLoader>;
+
+const messagesLoader = (locale: string): MessageLoader =>
+  resourceLoader(() =>
+    import(`../../translations/${locale}.json`).then(
+      (response: { default: ResolvedIntlConfig['messages'] }) => response.default,
+    ),
+  );
+
+const messageLoaders: MessageLoaders = VERNA_SUPPORTED_LOCALES.reduce(
+  (r: MessageLoaders, locale: string) => ({
+    ...r,
+    [locale]: messagesLoader(locale),
+  }),
+  {},
```

Comment:
Why "r"? Maybe a more explicit name? i'm not sure what it is, but if it's an objet [locale]: messages, won't it conflict if the locale declared bellow already exist in r? like overriding it entierly?